File: quickstart-java/app/src/main/java/com/hypertrack/quickstart/MyMessagingService.java
package com.hypertrack.quickstart;


import android.util.Log;

import com.google.firebase.messaging.RemoteMessage;
import com.hypertrack.sdk.HyperTrackMessagingService;

public class MyMessagingService extends HyperTrackMessagingService {
    private static final String TAG = "MyMessagingService";
    @Override
    public void onNewToken(String newToken) {
        super.onNewToken(newToken);
        Log.d(TAG, "Firebase token updated");
        // handle new token here
    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        if (!super.onMessageReceived(remoteMessage.getData())) {
            Log.d(TAG, "New push notification received");
            // process your push here
        } else {
            Log.d(TAG, "HyperTrack's data message was received");
            // you aren't require to do anything, but can log it for
            // debugging or other purpose
        }
    }
}

File: quickstart-kotlin/app/src/main/java/com/hypertrack/quickstart/MyMessagingService.kt
package com.hypertrack.quickstart

import android.util.Log
import com.google.firebase.messaging.RemoteMessage
import com.hypertrack.sdk.HyperTrackMessagingService

class MyMessagingService : HyperTrackMessagingService() {
    override fun onNewToken(newToken: String) {
        super.onNewToken(newToken)
        Log.d(TAG, "Firebase token has changed")
        // handle token update
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        if (!super.onMessageReceived(remoteMessage.data)) {
            // process notification
            Log.d(TAG, "Non-hypertrack push notification received")
        } else {
            Log.d(TAG, "hypertrack data message received")
            // that notification was processed by hypertrack sdk
            // You aren't required to do anything with it, as it was already processedd in
            // `super.onMessageReceived()`but you can log it for debugging or other purpose
        }

    }

    companion object {
        private const val TAG = "MyMessagingService"
    }
}